nmagent/client.go

```diff
@@ -0,0 +1,225 @@
+package nmagent
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/Azure/azure-container-networking/nmagent/internal"
+)
+
+// NewClient returns an initialized Client using the provided configuration
+func NewClient(host string, port uint16, opts ...Option) *Client {
+	client := &Client{
+		httpClient: &http.Client{
+			Transport: &internal.WireserverTransport{
+				Transport: http.DefaultTransport,
+			},
+		},
```

Comment:
I've been thinking more about this, and I think using a Wireserver Transport will make wireserver/nmagent issues more difficult for the team to debug. My main worry is that from this layer, it's not clear if the http status code came from wireserver or nmagent, while with the existing protocol we still have both in scope. I really appreciate the investigation and effort invested in `wireserver.go`, which would still be nice to keep for documentation purposes, but for now I'd suggest to keep `httpStatusCode` in the response structs and skip the transport approach. Thoughts @timraymond @rbtr?